Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -31,15 +31,91 @@ namespace AzToolsFramework
         }
     }
 
-    DPERowWidget::DPERowWidget(QWidget* parentWidget)
+    QSize DPELayout::sizeHint() const
+    {
+        int cumulativeWidth = 0;
+        int preferredHeight = 0;
+
+        // sizeHint for this horizontal layout is the sum of the preferred widths,
+        // and the maximum of the preferred heights
+        for (int layoutIndex = 0; layoutIndex < count(); ++layoutIndex)
+        {
+            QWidget* widgetChild = itemAt(layoutIndex)->widget();
+            if (widgetChild)
+            {
+                auto widgetSizeHint = widgetChild->sizeHint();
+                cumulativeWidth += widgetSizeHint.width();
+                preferredHeight = qMax(widgetSizeHint.height(), preferredHeight);
+            }
+        }
+        return { cumulativeWidth, preferredHeight };
+    }
+
+    QSize DPELayout::minimumSize() const
+    {
+        int cumulativeWidth = 0;
+        int minimumHeight = 0;
+
+        // minimumSize for this horizontal layout is the sum of the min widths,
+        // and the maximum of the preferred heights
+        for (int layoutIndex = 0; layoutIndex < count(); ++layoutIndex)
+        {
+            QWidget* widgetChild = itemAt(layoutIndex)->widget();
+            if (widgetChild)
+            {
+                const auto minWidth = widgetChild->minimumSizeHint().width();
+                ;
+                if (minWidth > 0)
+                {
+                    cumulativeWidth += minWidth;
+                }
+                minimumHeight = qMax(widgetChild->sizeHint().height(), minimumHeight);
+            }
+        }
+        return { cumulativeWidth, minimumHeight };
+    }
+
+    void DPELayout::setGeometry(const QRect& rect)
+    {
+        QLayout::setGeometry(rect);
+
+        // for now, just divide evenly horizontally. Later, we'll implement QSplitter-like
+        // functionality to allow the user to resize columns within a DPE
+        int perItemWidth = rect.width() / count();
```

Comment:
The default column division for the RPE was 2/5 - 3/5, as per the UX guidelines here: https://www.o3de.org/docs/tools-ui/ux-patterns/component-card/overview/

I'm not sure how simple it would be to replicate since we account for a variable amount of columns, but we should try to default to that.

Reply:
awesome! Thanks, I've never seen that. I rewrote it to special case 2 columns :)